openwisp-config/files/openwisp.agent

```diff
@@ -274,6 +279,7 @@ configuration_changed() {
 	local exit_code=$?
 
 	if [ "$exit_code" != "0" ]; then
+		cp "$PERSISTENT_CHECKSUM" "$CONFIGURATION_CHECKSUM"
```

Comment:
Is it because the get checksum operation can fail? And a proper checksum is restored?
Or for what? Can you please clarify? It may be good to add a comment to clarify this in the code and help our future selves remember this.

Reply:
Yes, this is to restore the last checksum if the get_checksum call fails.
I will add a comment and also an if statement to check, if $PERSISTENT_CHECKSUM exists.